### PR TITLE
expose more pulp parameters

### PR DESCRIPTION
**Pull request by timogoebel**
This PR exposed more pulp parameters so users have more choices what they want.
This is in preparation to manage pulp also for a pulp master by this module.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -26,6 +26,20 @@
 #
 # $pulp_max_speed::                     The maximum download speed per second for a Pulp task, such as a sync. (e.g. "4 Kb" (Uses SI KB), 4MB, or 1GB" )
 #
+# $pulp_max_tasks_per_child::           Number of tasks after which the worker is restarted and the memory it allocated is returned to the system
+#
+# $pulp_num_workers::                   Number of Pulp workers to use.
+#
+# $pulp_proxy_password::                Proxy password for authentication
```
Comment (ekohl) [inline manifests/init.pp]: I'd prefer this to be under username.

**Comment by timogoebel** [comment]
@ekohl: I finally found time to revisit this.

* rebased
* moved the parameter password parameter under the username parameter
* added a parameter for the ca_cert

File: manifests/init.pp [hunk 2]
```diff
@@ -20,6 +20,22 @@
...
+#
+# $pulp_proxy_url::                     URL of the http proxy server
+#
+# $pulp_proxy_username::                Proxy username for authentication
+#
+# $pulp_proxy_password::                Proxy password for authentication
+#
+# $pulp_puppet_wsgi_processes::         Number of WSGI processes to spawn for the puppet webapp
+#
+# $pulp_ca_cert::                       CA certificate for pulp
```
Comment (ekohl) [inline manifests/init.pp]: I feel this is a bit brief. The puppet-pulp description is: Absolute path to PEM encoded CA certificate file, used by Pulp to validate the identity of the broker using SSL.

File: manifests/init.pp [hunk 3]
```diff
@@ -52,6 +68,14 @@
...
   String $pulp_admin_password = $foreman_proxy_content::params::pulp_admin_password,
   Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+  Optional[Integer] $pulp_max_tasks_per_child = $foreman_proxy_content::params::pulp_max_tasks_per_child,
+  Optional[Integer] $pulp_num_workers = $foreman_proxy_content::params::pulp_num_workers,
+  Optional[String] $pulp_proxy_password = $foreman_proxy_content::params::pulp_proxy_password,
+  Optional[Integer] $pulp_proxy_port = $foreman_proxy_content::params::pulp_proxy_port,
+  Optional[String] $pulp_proxy_url = $foreman_proxy_content::params::pulp_proxy_url,
+  Optional[String] $pulp_proxy_username = $foreman_proxy_content::params::pulp_proxy_username,
+  Optional[Integer] $pulp_puppet_wsgi_processes = $foreman_proxy_content::params::pulp_puppet_wsgi_processes,
+  Optional[String] $pulp_ca_cert = $foreman_proxy_content::params::pulp_ca_cert,
```
Comment (ekohl) [inline manifests/init.pp]: This can be ```Optional[Stdlib::Absolutepath]```

File: manifests/init.pp [hunk 4]
```diff
@@ -52,6 +68,14 @@
   Boolean $pulp_master = $foreman_proxy_content::params::pulp_master,
   String $pulp_admin_password = $foreman_proxy_content::params::pulp_admin_password,
   Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+  Optional[Integer] $pulp_max_tasks_per_child = $foreman_proxy_content::params::pulp_max_tasks_per_child,
+  Optional[Integer] $pulp_num_workers = $foreman_proxy_content::params::pulp_num_workers,
+  Optional[String] $pulp_proxy_password = $foreman_proxy_content::params::pulp_proxy_password,
+  Optional[Integer] $pulp_proxy_port = $foreman_proxy_content::params::pulp_proxy_port,
```
Comment (ekohl) [inline manifests/init.pp]: This can limit to ```[0, 65535]```

File: manifests/init.pp [hunk 5]
```diff
@@ -52,6 +68,14 @@
   Boolean $pulp_master = $foreman_proxy_content::params::pulp_master,
   String $pulp_admin_password = $foreman_proxy_content::params::pulp_admin_password,
   Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+  Optional[Integer] $pulp_max_tasks_per_child = $foreman_proxy_content::params::pulp_max_tasks_per_child,
```
Comment (ekohl) [inline manifests/init.pp]: Should this be at least 1?

File: manifests/init.pp [hunk 6]
```diff
@@ -52,6 +68,14 @@
...
+  Optional[Integer] $pulp_num_workers = $foreman_proxy_content::params::pulp_num_workers,
```
Comment (ekohl) [inline manifests/init.pp]: Should this be at least 1?

File: manifests/init.pp [hunk 7]
```diff
@@ -52,6 +68,14 @@
...
+  Optional[String] $pulp_proxy_password = $foreman_proxy_content::params::pulp_proxy_password,
+  Optional[Integer] $pulp_proxy_port = $foreman_proxy_content::params::pulp_proxy_port,
+  Optional[String] $pulp_proxy_url = $foreman_proxy_content::params::pulp_proxy_url,
+  Optional[String] $pulp_proxy_username = $foreman_proxy_content::params::pulp_proxy_username,
+  Optional[Integer] $pulp_puppet_wsgi_processes = $foreman_proxy_content::params::pulp_puppet_wsgi_processes,
```
Comment (ekohl) [inline manifests/init.pp]: Should this be at least 1?

**Comment by timogoebel** [comment]
@ekohl: Addressed your comments, thanks.

File: manifests/init.pp [hunk 8]
```diff
@@ -52,6 +68,14 @@
   Boolean $pulp_master = $foreman_proxy_content::params::pulp_master,
   String $pulp_admin_password = $foreman_proxy_content::params::pulp_admin_password,
   Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+  Optional[Integer[1,]] $pulp_max_tasks_per_child = $foreman_proxy_content::params::pulp_max_tasks_per_child,
```
Comment (ekohl) [inline manifests/init.pp]: I'm surprised the trailing comma works but I think it's cleaner without in this case.

**Comment by timogoebel** [comment]
@ekohl: Done.

File: manifests/init.pp [hunk 9]
```diff
@@ -52,6 +68,14 @@
   Boolean $pulp_master = $foreman_proxy_content::params::pulp_master,
   String $pulp_admin_password = $foreman_proxy_content::params::pulp_admin_password,
   Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+  Optional[Integer[1]] $pulp_max_tasks_per_child = $foreman_proxy_content::params::pulp_max_tasks_per_child,
+  Optional[Integer[1]] $pulp_num_workers = $foreman_proxy_content::params::pulp_num_workers,
```
Comment (stbenjam) [inline manifests/init.pp]: Shouldn't these get passed to pulp class?

**Comment by timogoebel** [comment]
@stbenjam: Thanks, added the missing params.

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**